extensions/tasks-parent/tasks-client/src/main/java/io/resys/thena/tasks/client/spi/visitors/TaskCommandVisitor.java

```diff
@@ -111,28 +144,161 @@ private Task visitCreateTask(CreateTask command) {
+    return this.current;
+  }
+
+  private Task visitAssignTask(AssignTask command) {
+    this.current = this.current
+        .withAssigneeIds(command.getAssigneeIds().stream().distinct().sorted().toList())
+        .withUpdated(handleTargetDate(command.getTargetDate()));
+    return this.current;
+  }
+
+  private Task visitChangeTaskDueDate(ChangeTaskDueDate command) {
+    this.current = this.current
+        .withDueDate(command.getDueDate().orElse(null))
+        .withUpdated(handleTargetDate(command.getTargetDate()));
+    return this.current;
+  }
+
+  private Task visitChangeTaskInfo(ChangeTaskInfo command) {
+    this.current = this.current
+        .withTitle(command.getTitle())
+        .withDescription(command.getDescription())
+        .withUpdated(handleTargetDate(command.getTargetDate()));
+    return this.current;
+  }
+
+  private Task visitCreateTaskExtension(CreateTaskExtension command) {
+    final var extensions = new ArrayList<>(current.getExtensions());
+    final var id = ctx.getGid().getNextId(DocumentType.TASK);
+    extensions.add(ImmutableTaskExtension.builder()
+        .id(id)
+        .name(command.getName())
+        .type(command.getType())
+        .body(command.getBody())
+        .build());
+    this.current = this.current
+        .withExtensions(extensions.stream().toList())
+        .withUpdated(handleTargetDate(command.getTargetDate()));
+    return this.current;
+  }
+
+  private Task visitChangeTaskExtension(ChangeTaskExtension command) {
+    final var id = command.getId();
+
+    final var newExtension = ImmutableTaskExtension.builder()
+        .id(id)
+        .name(command.getName())
+        .type(command.getType())
+        .body(command.getBody())
+        .build();
+
+    final var newExtensionList = visitList(current.getExtensions(), newExtension, id, Task.TaskExtension::getId, "Extension");
+
+    this.current = this.current
+        .withExtensions(newExtensionList.stream().sorted(Comparator.comparing(Task.TaskExtension::getId)).toList())
+        .withUpdated(handleTargetDate(command.getTargetDate()));
+    return this.current;
+  }
+
+  private <T, P> List<P> visitList(final List<P> currentItems, final P newItem, String currentId,  final Function<P, T> getter, String name) {
+    final var newItems = new ArrayList<P>();
+    boolean found = false;
+    for (final var item : currentItems) {
+      if (getter.apply(item).equals(currentId)) {
+        newItems.add(newItem);
+        found = true;
+      } else {
+        newItems.add(item);
+      }
+    }
+    if (!found) {
+      throw new UpdateTaskVisitorException(String.format(name + " with id %s not found", currentId));
+    }
+    return newItems.stream().toList();
+  }
+
+  private Task visitAssignTaskParent(AssignTaskParent command) {
+    this.current = this.current
+        .withParentId(command.getParentId())
+        .withUpdated(handleTargetDate(command.getTargetDate()));
+    return this.current;
+  }
+
```

Comment:
1. This is not `handleTargetDate`.  This is `requireTargetDate`.
2. You don't need a try/catch block for checking if object is null or not.  Just write a plain `if` here.  If object is null, then throw `UpdateTaskVisitorException`.